### ISSUE TITLE
add source-map-support attribute to the nodejs_binary rule

### ISSUE THREAD

**Issue by IgorMinar** [bazelbuild/rules_nodejs#389]
Currently we expect users to either provide the `source-map-support` npm package as a label to the `data` attribute of the nodejs_binary rule, or disable the source-map-support via `install_source_map_support` attribute which can be set to false. If neither is done, a warning is issued by the nodejs_binary, which is hard to understand or action upon:

```
WARNING: source-map-support module not installed.
    Stack traces from languages like TypeScript will point to generated .js files.
    Set install_source_map_support = False in //packages/bazel/src/ngc-wrapped:ngc-wrapped_bin to turn off this warning.
```

If we added an attribute to the `nodejs_binary` rule to enable the support by providing the needed npm package, we'd make the api more discoverable and intuitive and also got rid of the hard to understand warning. Example usage:

```
nodejs_binary(
    name = "ngc-wrapped",
    data = [
        ":ngc_lib",
        "@build_bazel_rules_typescript//third_party/github.com/bazelbuild/bazel/src/main/protobuf:worker_protocol.proto",
        "@ngdeps//tslib",
    ],
    entry_point = "angular/packages/bazel/src/ngc-wrapped/index.js",
    source_map_support = "@ngdeps//source-map-support",
    visibility = ["//visibility:public"],
)
```

**Comment by alexeagle** [comment]
One issue here is how it would work for users on coarse-grained deps.

**Comment by IgorMinar** [comment]
Accept a boolean in addition to a label? But do we still want to support both kinds of deps?

**Comment by alexeagle** [comment]
Yes, I think we still want to have the easy on-ramp that follows typical npm semantics so that use of Bazel doesn't always mean having to re-declare your npm dependencies everywhere.

**Comment by alexeagle** [comment]
@vikerman fixed this by vendoring in the source-map-support files to break the runtime dependency on loading them.
